app/utils/checklist.js

```diff
@@ -26,43 +28,147 @@ export default class ChecklistUtil {
+    };
+  }
+
+  /**
+   * This function returns the languages of an asset and its children recursively
+   * @param {object} asset The asset to find the languages of
+   * @param {object} languages Empty object that acts like a map to store the languages found as keys
+   * @returns {array} An array containing the languages found
+   */
+  static findAssetLanguages(asset, languages = {}) {
+    if (
+      asset.type === Constants.AssetType.FILE &&
+      asset.contentTypes.includes(Constants.AssetContentType.CODE)
+    ) {
       const lastSep = asset.uri.lastIndexOf(path.sep);
       const fileName = asset.uri.substring(lastSep + 1);
       const ext = fileName.split('.').pop();
 
-      if(ext){
+      if (ext) {
         AssetsConfig.contentTypes.forEach((contentType) => {
           // Ensures both the extension and content type are for code files
-          if(contentType.categories.includes(Constants.AssetContentType.CODE) && contentType.extensions.includes(ext)) {
+          if (
+            contentType.categories.includes(Constants.AssetContentType.CODE) &&
+            contentType.extensions.includes(ext)
+          ) {
             languages[contentType.name] = true;
           }
         });
       }
     }
 
-    if(asset.children){
-      asset.children.forEach(child => {
-        this.findAssetLanguagesAndDependencies(child, languages, dependencies);
+    if (asset.children) {
+      asset.children.forEach((child) => {
+        ChecklistUtil.findAssetLanguages(child, languages);
       });
     }
 
-    return {
-      languages: Object.keys(languages),
-      dependencies: Object.keys(dependencies)
-    };
+    return Object.keys(languages);
+  }
+
+  /**
+   * This function returns the dependencies of an asset and its children recursively
+   * @param {object} asset The asset to find the dependencies of
+   * @param {object} dependencies Empty object that acts like a map to store the dependencies found as keys
+   * @returns {array} An array containing the dependencies found
+   */
+  static findAssetDependencies(asset) {
+    const dependencies = [];
+    const assetDepedencies = WorkflowUtil.getAllLibraryDependencies(asset);
+    assetDepedencies.forEach((x) => {
+      if (x.assetType && x.assetType !== Constants.AssetType.GENERIC) {
+        x.dependencies.forEach((dep) => {
+          if (dependencies.findIndex((i) => i === dep.id) === -1) {
+            dependencies.push(WorkflowUtil.getShortDependencyName(dep.id));
+          }
+        });
+      }
+    });
+    return dependencies;
+  }
+
+  /** This function finds the data files in the asset and its children recursively
+   * @param {object} asset The asset to find the data files within
+   * @param {array} dataFiles An array to store the data files found
+   * @returns {object} An object containing the data files found
+   */
+  static findDataFiles(asset, dataFiles = []) {
+    if (!asset) {
+      return { datafiles: dataFiles };
```

Comment:
Could we capitalize "F" and make it `dataFiles` throughout?

Reply:
Actually that first one is what is also used for text that's displayed in UI, for each scan result title. If camelcase for title works for you? I mean all that UI is in a very 'codey' font, so no problem.

Reply:
I'd be okay with camel case for these - thanks!